Tests/UnitTests/Core/Geometry/ConeVolumeBoundsTests.cpp

```diff
@@ -0,0 +1,90 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Geometry/ConeVolumeBounds.hpp"
+#include "Acts/Tests/CommonHelpers/FloatComparisons.hpp"
+#include "Acts/Tests/CommonHelpers/ObjTestWriter.hpp"
+#include "Acts/Utilities/BoundingBox.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Units.hpp"
+
+namespace tt = boost::test_tools;
+
+namespace Acts {
+
+using namespace UnitLiterals;
+
+// Create a test context
+GeometryContext tgContext = GeometryContext();
+
+namespace Test {
+
+BOOST_AUTO_TEST_SUITE(VolumeBounds)
+
+BOOST_AUTO_TEST_CASE(ConeVolumeBoundsTests) {
+  std::vector<IdentifiedPolyhedron> tPolyhedrons;
+
+  // Single solid Cone
+  ConeVolumeBounds solidCone(0., 0., 0.45, 50_mm, 50_mm, 0., M_PI);
+
+  // Test correct parameter return
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eInnerAlpha) == 0.);
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eInnerOffsetZ) == 0.);
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eOuterAlpha) == 0.45);
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eOuterOffsetZ) == 50.);
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eHalfLengthZ) == 50.);
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eAveragePhi) == 0.);
+  BOOST_TEST(solidCone.get(ConeVolumeBounds::eHalfPhiSector) == M_PI);
+  // Derived quantities
+  BOOST_TEST(solidCone.innerTanAlpha() == 0.);
+  BOOST_TEST(solidCone.innerRmin() == 0.);
+  BOOST_TEST(solidCone.innerRmax() == 0.);
+  BOOST_TEST(solidCone.outerTanAlpha() == std::tan(0.45));
+
+  double outerRmax = 100_mm * solidCone.outerTanAlpha();
+  BOOST_TEST(solidCone.outerRmin() == 0.);
+  BOOST_TEST(solidCone.outerRmax() == outerRmax);
+
+  auto solidConeSurfaces = solidCone.orientedSurfaces();
+  BOOST_TEST(solidConeSurfaces.size() == 2);
+
+  // Single solid Cone - with cut off
+  ConeVolumeBounds cutOffCone(0., 0., 0.45, 80_mm, 50_mm, 0., M_PI);
+  auto cutOffConeSurfaces = cutOffCone.orientedSurfaces();
+  BOOST_TEST(cutOffConeSurfaces.size() == 3);
```

Comment:
I would also test the resulting surfaces here to make sure that they are correctly oriented, e.g. by comparing normal vectors.

Reply:
Yes, this has been requested already by @FabianKlimpel - for the remaining VolumeBounds and implemented in #196 - as this was a parallel development, I will add it here as well.

Reply:
Done.